contracts/lpp/src/state/config.rs

```diff
@@ -53,41 +59,50 @@ impl Config {
         Self::STORAGE.load(storage).map_err(Into::into)
     }
 
-    pub fn update_lease_code(storage: &mut dyn Storage, lease_code: Uint64) -> Result<()> {
-        Self::STORAGE
-            .update::<_, ContractError>(storage, |mut config| {
-                config.lease_code_id = lease_code;
-
-                Ok(config)
-            })
-            .map(|_| ())
-    }
-
-    pub fn update_borrow_rate(storage: &mut dyn Storage, borrow_rate: InterestRate) -> Result<()> {
-        Self::STORAGE
-            .update(storage, |mut config| {
-                config.borrow_rate = borrow_rate;
-
-                Ok(config)
-            })
-            .map(|_| ())
-    }
-
-    pub fn initial_derivative_price<LPN>() -> Price<NLpn, LPN>
+    pub fn initial_derivative_price<Lpn>() -> Price<NLpn, Lpn>
     where
-        LPN: Currency + Serialize + DeserializeOwned,
+        Lpn: Currency + Serialize + DeserializeOwned,
     {
         Price::identity()
     }
 }
 
+macro_rules! update_param_impl {
+    ($($method:ident($field:ident : $type:ty)),+ $(,)?) => {
+        impl Config {
+            $(
+                pub fn $method(
+                    storage: &mut dyn Storage,
+                    $field: $type,
+                ) -> Result<()> {
+                    Self::STORAGE
+                        .update(storage, |config: Self| {
+                            Ok(Self {
+                                $field,
+                                ..config
+                            })
+                        })
+                        .map(|_| ())
+                }
+            )+
+        }
+    };
+}
+
+update_param_impl!(
+    update_lease_code(lease_code_id: Uint64),
+    update_borrow_rate(borrow_rate: InterestRate),
+    update_min_utilization(min_utilization: BoundToHundredPercent),
+);
+
```

Comment:
we have discussed many times that macro-ses are not welcome where we may provide functions, please rework

Reply:
While I do agree, we also have discussed that code duplication is unacceptable too, and that's a lot of it.